core/src/main/scala/org/apache/spark/serializer/SerializerManager.scala

```diff
@@ -103,6 +107,20 @@ private[spark] class SerializerManager(defaultSerializer: Serializer, conf: Spar
   }
 
   /**
+   * Wrap an input stream for encryption if shuffle encryption is enabled
+   */
+  def wrapForEncryption(s: InputStream): InputStream = {
```

Comment:
Doing this would also simplify other interfaces, e.g., you wouldn't need the dummy `EncryptStream` classes in a bunch of tests because `DiskBlockObjectWriter` would just take a generic `wrapStream` function instead of separate ones for compression and encryption.